### PR TITLE
Add Authenticated User to params

### PR DESCRIPTION
**Pull request by megajunky**
Until now, the User data was only available in the Session. I added a Middleware to add the User data also to the params for easier access in routes. 

Maybe you like it.

### CODE REVIEW

**Comment by rappet** [comment]
I don't like adding it to the params, as the params do usualy mean that they are some kind of data provided for that _specific_ request by the user; the user name/id/email is neither provided by the user nor specific for this single request.

**Comment by rappet** [comment]
> I don't like adding it to the params, as the params do usualy mean that they are some kind of data provided for that _specific_ request by the user; the user name/id/email is neither provided by the user nor specific for this single request.

How is this easier accessable? Isn't it just a different key to look for?

File: src/clj/auth.clj [hunk 1]
```diff
@@ -85,12 +85,11 @@
   [request]
   (let [github-data (github-user-data-from-token (extract-token request))]
     {:status 200
-     :headers {"Content-Type" "text/plain"}
-     :session (merge (:session request) github-data)
+     :session (assoc (:session request) :user github-data)
```
Comment (rappet) [inline src/clj/auth.clj:88]: I really like that.
Better than my `:user-id`, `:user-email` and `:user-name` keys!

**Comment by rappet** [comment]
Please don't close that MR. It has nice things in it.

**Comment by megajunky** [comment]
Ok 😄

**Comment by rappet** [comment]
The correct way to do this btw. is to add comments or code-suggestions to the original MR.
Also if you get comments on something you should give an answar instead of just closing that MR :)

**Comment by rappet** [comment]
If you don't mind I will just copy the `:user`-thinggy to the original MR, as this is better to clean up than to merge into an MR into another still-WIP MR.

**Comment by megajunky** [comment]
Yeah you are right. I thought we don't need this PR anymore.

**Comment by megajunky** [comment]
> How is this easier accessable? Isn't it just a different key to look for?

Yes. It’s just another path. I think you are right and it's a bit confusing if the user just pops up in the params.

**Comment by rappet** [comment]
@megajunky see https://github.com/pkoerner/expert-parakeet/pull/56

**Comment by megajunky** [comment]
Changes are in #8

**Pull request closed without merging**